include/fence.h

```diff
@@ -137,16 +132,33 @@ std::future<buffer_snapshot<DataT, Dims>> distr_queue::fence(const buffer<DataT,
 	return future;
 }
 
-} // namespace celerity
+} // namespace celerity::detail
+
+namespace celerity {
+class distr_queue;
+}
 
 namespace celerity::experimental {
 
 template <typename T, int Dims>
 using buffer_snapshot [[deprecated("buffer_snapshot is no longer experimental, use celerity::buffer_snapshot")]] = celerity::buffer_snapshot<T, Dims>;
 
-template <typename... Params>
-[[deprecated("fence is no longer experimental, use celerity::distr_queue::fence")]] [[nodiscard]] auto fence(celerity::distr_queue& q, const Params&... args) {
-	return q.fence(args...);
+template <typename T, int Dims>
+[[deprecated("fence is no longer experimental, use celerity::queue::fence")]] [[nodiscard]] auto fence(
+    celerity::distr_queue& /* q */, const buffer<T, Dims>& buf, const subrange<Dims>& sr) {
```

Comment:
Any particular reason for spelling the overloads out explicitly? (Not that I'm against it, just curious)

Reply:
Reason is backwards compatibility: The explicit parameter types allow the second argument of `experimental::fence(buf, {{0, 0}, {10, 10}})` to be deduced as a `subrange<2>`, but that would not be possible on `fence(Params &&...args)`.